File: Codeforces Round 939 (Div. 2)/1.cpp
def max_points(n, a, b):
    a.sort(reverse=True)
    b.sort(reverse=True)
    i = j = points = 0
    
    while i < len(a) and j < len(b):
        if a[i] > b[j]:
            points += 1
            i += 1
            j += 1
        else:
            j += 1
    
    return points

# Read input
t = int(input())
for _ in range(t):
    n = int(input())
    a = list(map(int, input().split()))
    b = list(map(int, input().split()))
    
    print(max_points(n, a, b))

File: Codeforces Round 939 (Div. 2)/1.py
def find_winners(k, q, a, n_list):
    results = []
    for n in n_list:
        players = list(range(1, n+1))
        while True:
            to_remove = [players[i-1] for i in a if i <= len(players)]
            if not to_remove:
                break
            players = [player for player in players if player not in to_remove]
        results.append(len(players))
    return results

test_cases = []

t = int(input().strip())
for _ in range(t):
    k, q = map(int, input().split())
    a = list(map(int, input().split()))
    n_list = list(map(int, input().split()))
    test_cases.append((k, q, a, n_list))

outputs = []

for case in test_cases:
    winners = find_winners(*case)
    outputs.append(winners)

for winners in outputs:
    print(' '.join(map(str, winners)))
